src/javascript/app/pages/user/new_account/welcome_page.js

```diff
@@ -1,45 +1,59 @@
 const BinarySocket     = require('../../../base/socket');
 const Client           = require('../../../base/client');
 const getElementById   = require('../../../../_common/common_functions').getElementById;
-const localize         = require('../../../../_common/localize').localize;
 const Url              = require('../../../../_common/url');
-const createElement    = require('../../../../_common/utility').createElement;
 const showLoadingImage = require('../../../../_common/utility').showLoadingImage;
+const ClientBase       = require('../../../../_common/base/client_base');
+const State            = require('../../../../_common/storage').State;
 
 const WelcomePage = (() => {
+
+    let el_welcome_container,
+        is_virtual,
+        upgrade_info,
+        is_uk,
+        is_unwelcome_uk;
+
+    const init = () => {
+        upgrade_info          = ClientBase.getBasicUpgradeInfo();
+        is_virtual            = Client.get('is_virtual');
+        el_welcome_container  = getElementById('welcome_container');
+        is_uk                 = Boolean(Client.get('residence') === 'gb');
+        is_unwelcome_uk       = State.getResponse('get_account_status.status').some(status => status === 'unwelcome') && (/gb/.test(Client.get('residence')));
+    };
+    
+    const getCanUpgrade = (upgrade_type ,  { can_upgrade_to } = upgrade_info) => can_upgrade_to.includes(upgrade_type);
+  
     const onLoad = () => {
-        BinarySocket.wait('authorize', 'landing_company', 'get_settings').then(() => {
-            const el_welcome_container = getElementById('welcome_container');
+        BinarySocket.wait('authorize', 'landing_company', 'get_settings', 'get_account_status').then(() => {
+            init();
             if (Client.hasAccountType('real')) {
                 window.location.href = Client.defaultRedirectUrl();
                 showLoadingImage(el_welcome_container, 'dark');
             }
 
-            const upgrade_info = Client.getUpgradeInfo();
-            const el_upgrade_title = getElementById('upgrade_title');
-            let upgrade_btn_txt = '';
+            getElementById('default').addEventListener('click', () => {
```

Comment:
why does something have the generic id `default` 🤔 that sounds like a really bad id to have 😄